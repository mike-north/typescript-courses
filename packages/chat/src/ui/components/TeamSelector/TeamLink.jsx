import * as React from 'react'
import { Link, useRouteMatch } from 'react-router-dom'

const TeamLink = ({ team }) => {
  const match = useRouteMatch({
    path: `/team/${team.id}`,
    exact: false,
  })

  return (
    <Link
      to={`/team/${team.id}/channel/${team.channels[0].id}`}
      className={
        'team-selector__team-button cursor-pointer rounded-lg p-2 pl-4 inline-block sm:block no-underline opacity-25 ' +
        (match ? 'opacity-100' : '')
      }
    >
      <div className="bg-white h-12 w-12 flex items-center justify-center text-black text-2xl font-semibold rounded-lg mb-1 overflow-hidden">
        <img
          className="team-selector__team-logo"
          src={team.iconUrl}
          alt={`Join the ${team.name} chat`}
        />
      </div>
    </Link>
  )
}

export default TeamLink
