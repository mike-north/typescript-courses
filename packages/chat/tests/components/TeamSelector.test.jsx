import * as React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import * as renderer from 'react-test-renderer'
import TeamSelector from '../../src/ui/components/TeamSelector'

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Router>
      <TeamSelector
        teams={[
          {
            name: 'LinkedIn',
            id: 'linkedin',
            iconUrl: '',
            channels: [
              {
                name: 'general',
                id: 'g',
                teamId: 'linkedin',
                description: 'general chat for general folks',
              },
            ],
          },
        ]}
      />
    </Router>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
