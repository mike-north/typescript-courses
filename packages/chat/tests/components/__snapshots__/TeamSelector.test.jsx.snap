// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<nav
  className="team-selector bg-indigo-900 border-indigo-900 border-r-2 pt-2 text-purple-300 flex-none block"
>
  <a
    className="team-selector__team-button cursor-pointer rounded-lg p-2 pl-4 inline-block sm:block no-underline opacity-25 "
    href="/team/linkedin/channel/g"
    onClick={[Function]}
  >
    <div
      className="bg-white h-12 w-12 flex items-center justify-center text-black text-2xl font-semibold rounded-lg mb-1 overflow-hidden"
    >
      <img
        alt="Join the LinkedIn chat"
        className="team-selector__team-logo"
        src=""
      />
    </div>
  </a>
  <div
    className="team-selector__add-team-button cursor-pointer p-4 inline-block sm:block"
  >
    <div
      className="bg-white opacity-25 h-12 w-12 flex items-center justify-center text-black text-2xl font-semibold rounded-lg mb-1 overflow-hidden"
    >
      <svg
        aria-hidden="true"
        className="fill-current h-10 w-10 block"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 10c0 .553-.048 1-.601 1H11v4.399c0 .552-.447.601-1 .601-.553 0-1-.049-1-.601V11H4.601C4.049 11 4 10.553 4 10c0-.553.049-1 .601-1H9V4.601C9 4.048 9.447 4 10 4c.553 0 1 .048 1 .601V9h4.399c.553 0 .601.447.601 1z"
        />
      </svg>
    </div>
  </div>
</nav>
`;
